### ISSUE TITLE
GRU内存过大

### ISSUE THREAD

**Issue by 2017040264** [PaddlePaddle/PaddleNLP#237]
我使用相同的代码，在AI STUDIO上跑的时候，GRU占用的内存非常大，导致项目根本跑不通。而在自己电脑上跑的时候，GRU占用的内存很小。所有我怀疑是平台出现了某个bug.
代码：

`
import paddle

class Seq2SeqEncoder(paddle.nn.Layer):
    def __init__(self, vocab_size, embed_dim, hidden_size):
        super(Seq2SeqEncoder, self).__init__()
        self.embedder = paddle.nn.Embedding(vocab_size, embed_dim)
        self.lstm = paddle.nn.LSTM(input_size=embed_dim,hidden_size=hidden_size,dropout=0.2 )
        
    def forward(self, sequence):
        inputs = self.embedder(sequence)
        encoder_output, encoder_state = self.lstm(inputs)
        return encoder_output, encoder_state

class Seq2SeqEncoder(paddle.nn.Layer):
    def __init__(self, vocab_size, embed_dim, hidden_size):
        super(Seq2SeqEncoder, self).__init__()
        self.embedder = paddle.nn.Embedding(vocab_size, embed_dim)
        self.lstm = paddle.nn.LSTM(input_size=embed_dim,hidden_size=hidden_size,dropout=0.2 )
        
    def forward(self, sequence):
        inputs = self.embedder(sequence)
        encoder_output, encoder_state = self.lstm(inputs)
        return encoder_output, encoder_state

class Encoder(paddle.nn.Layer):
    def __init__(self, vocab_size, embed_dim, hidden_size,rate=0.2):
        super(Encoder, self).__init__()
        self.embedder = paddle.nn.Embedding(vocab_size, embed_dim)
        self.gru = paddle.nn.GRU(input_size=embed_dim,hidden_size=hidden_size,dropout=rate)

    def forward(self, sequence):
        inputs = self.embedder(sequence)
        encoder_output, encoder_state = self.gru(inputs)
        return encoder_output, encoder_state
encoder=Encoder(vocab_size=30000,embed_dim=256,hidden_size=256)
paddle.summary(encoder,(128,200),dtypes='int64')`
本地运行截图：

![1](https://user-images.githubusercontent.com/57398559/113878301-0fe9f900-97ec-11eb-95ee-b8d5790a5e73.jpg)

平台运行截图：
![2](https://user-images.githubusercontent.com/57398559/113879175-d796ea80-97ec-11eb-90bc-46eb3b82227e.jpg)

[AI Studio 项目链接](https://aistudio.baidu.com/aistudio/projectdetail/1770318)

**Comment by joey12300** [comment]
@2017040264 您好，请问你自己电脑上和AI studio上的Paddle版本分别是多少呢？

**Comment by joey12300** [comment]
目前我跑了您提供的程序，是可以正常运行网络前后向。这边分别使用2.0.1和2.1.0.dev版本的paddle进行测试，结果是2.0.1显示很大的显存，而2.1.0.dev显示正常的显存，推测是2.0.1 paddle.summary有bug，显存计算有误。
2.0.1
![image](https://user-images.githubusercontent.com/10826371/113984519-c8ac4880-987d-11eb-8a11-25b596cbdd44.png)
2.1.0.dev
![image](https://user-images.githubusercontent.com/10826371/113984543-cea22980-987d-11eb-927e-5a75f4777854.png)

**Comment by ZHUI** [comment]
同issues https://github.com/PaddlePaddle/Paddle/issues/32140  

目前develop版本已修复。 https://github.com/PaddlePaddle/Paddle/issues/32140#issuecomment-817516098

**Comment by 2017040264** [comment]
好的，感谢。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jack ***@***.***&gt;; 
发送时间: 2021年4月8日(星期四) 下午3:23
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PaddlePaddle/PaddleNLP] GRU内存过大 (#237)





 
目前我跑了您提供的程序，是可以正常运行网络前后向。这边分别使用2.0.1和2.1.0.dev版本的paddle进行测试，结果是2.0.1显示很大的显存，而2.1.0.dev显示正常的显存，推测是2.0.1 paddle.summary有bug，显存计算有误。
 2.0.1
 
 2.1.0.dev
